### ISSUE TITLE
[4.0] JFormField FileList & FolderList have conflicting 'filter' attributes

### ISSUE THREAD

**Issue by bassmanpaul** [joomla/joomla-cms#14899]
So in JForm usually a field can have a `filter` attribute like `filter="cmd"` which allows JForm to filter the users input for that field. However, in the `FileList` & `FolderList` fields the `filter` attribute is overwritten and used in the field itself rather than by JForm.

In these fields the `filter` attribute is passed to `JFolder::files( $path, $this->filter )`. This means we can no longer filter the actual form field itself.

Is this intentional or is it an oversight?

If it is an oversight then the attribute that filters `JFolder` needs changing.

The fields in question are:

`libraries/joomla/form/fields/filelist.php`
`libraries/joomla/form/fields/folderlist.php`

I'm happy to create a pull request if you like, changing `filter=".css"` to `filefilter=""` perhaps?

This would be a breaking change as there are a number of Joomla forms that use this field that will need the attribute changing as well:

```
administrator/components/com_languages/models/forms/language.xml
     filter="\.gif$"
components/com_mailto/mailto.xml
     filter="."
plugins/editors/codemirror/codemirror.xml
     filter="\.css$"
plugins/editors/tinymce/form/setoptions.xml
     filter="\.js$"
```

**Comment by mbabker** [comment]
Added to the 4.0 task board.  This is a B/C break that is IMO going to be needed.

**Comment by franz-wohlkoenig** [comment]
Sorry, wrong Button.

**Comment by SharkyKZ** [comment]
Test PR #24977 please.

**Comment by ghost** [comment]
Closed as having Pull Request #24977
